src/NUnitFramework/framework/Internal/Commands/ConstructFixtureCommand.cs

```diff
@@ -18,19 +20,14 @@ public ConstructFixtureCommand(TestCommand innerCommand)
         {
             Guard.ArgumentValid(Test is TestSuite, "ConstructFixtureCommand must reference a TestSuite", nameof(innerCommand));
 
-            BeforeTest = (context) =>
+            BeforeTest = context =>
             {
-                ITypeInfo typeInfo = Test.TypeInfo;
+                ITypeInfo? typeInfo = Test.TypeInfo;
 
-                if (typeInfo != null)
-                {
+                if (typeInfo is {IsStaticClass: false})
                     // Use preconstructed fixture if available, otherwise construct it
-                    if (!typeInfo.IsStaticClass)
-                    {
-                        context.TestObject = Test.Fixture ?? typeInfo.Construct(((TestSuite)Test).Arguments);
-                        if (Test.Fixture == null)
-                            Test.Fixture = context.TestObject;
-                    }
+                {
+                    context.TestObject = Test.Fixture ??= typeInfo.Construct(((TestSuite)Test).Arguments);
```

Comment:
It took me a bit to think through the possibilities here to ensure this was still the same logic but it seems sound